src/main/java/io/github/mzmine/gui/chartbasics/graphicsexport/GraphicsExportDialogFX.java

```diff
@@ -0,0 +1,202 @@
+package io.github.mzmine.gui.chartbasics.graphicsexport;
+
+import java.awt.Color;
+import java.io.File;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.jfree.chart.JFreeChart;
+import org.jfree.chart.fx.ChartViewer;
+import org.jfree.chart.plot.DefaultDrawingSupplier;
+import org.jfree.chart.plot.DrawingSupplier;
+import org.jfree.chart.plot.XYPlot;
+import org.jfree.data.xy.XYDataset;
+import io.github.mzmine.gui.chartbasics.ChartLogicsFX;
+import io.github.mzmine.gui.chartbasics.chartthemes.ChartThemeFactory;
+import io.github.mzmine.gui.chartbasics.chartthemes.ChartThemeParameters;
+import io.github.mzmine.gui.chartbasics.chartthemes.EStandardChartTheme;
+import io.github.mzmine.gui.chartbasics.gui.javafx.EChartViewer;
+import io.github.mzmine.main.MZmineCore;
+import io.github.mzmine.parameters.Parameter;
+import io.github.mzmine.parameters.ParameterSet;
+import io.github.mzmine.parameters.dialogs.ParameterSetupDialog;
+import io.github.mzmine.parameters.parametertypes.DoubleComponent;
+import io.github.mzmine.parameters.parametertypes.DoubleParameter;
+import io.github.mzmine.parameters.parametertypes.FontParameter;
+import io.github.mzmine.parameters.parametertypes.FontSpecs;
+import io.github.mzmine.parameters.parametertypes.OptionalParameterComponent;
+import io.github.mzmine.util.color.Colors;
+import javafx.scene.control.Button;
+import javafx.scene.layout.BorderPane;
+
+public class GraphicsExportDialogFX extends ParameterSetupDialog {
+
+  private static final Logger logger = Logger.getLogger(GraphicsExportDialogFX.class.getName());
+
+  protected EStandardChartTheme theme;
+  protected BorderPane pnChartPreview;
+  protected JFreeChart chart;
+  protected EChartViewer chartPanel;
+  protected ChartThemeParameters chartParam;
+  protected Color[] colors;
+
+  private Button btnRenewPreview;
+  private Button btnApply;
+  private Button btnSave;
+
+
+  public GraphicsExportDialogFX(boolean valueCheckRequired, ParameterSet parameterSet,
+      JFreeChart chart) {
+    super(valueCheckRequired, parameterSet);
+
+    chartParam = (ChartThemeParameters) parameterSet

```

Comment:
No need to cast parameterSet to ChartThemeParameters here - the getParameter() method is universal to all ParameterSet classes